msteams-platform/messaging-extensions/build-api-based-message-extension.md

```diff
@@ -8,513 +8,28 @@ ms.author: anclear
 ms.date: 10/19/2023
 ---
 
-# Build API-based message extension
+# Enable authentication for API-based message extensions
 
-> [!NOTE]
-> API-based message extensions only support search commands.
+Authentication is a fundamental aspect of security and serves as the first line of defense, ensuring that access to systems, applications, and data is granted only to those with verified credentials. Authentication for API-based message extensions is crucial for several reasons:
 
-API-based message extensions are a Microsoft Teams app capability that integrates external APIs directly into Teams, enhancing your app's usability and offering a seamless user experience. API-based message extensions support search commands and can be used to fetch and display data from external services within Teams, streamlining workflows by reducing the need to switch between applications.
+* **Security**: It protects against unauthorized access and potential breaches, safeguarding both user data and the integrity of the system.
+* **Data Privacy**: Ensures that personal and sensitive information is only accessible to users with the correct permissions.
 
-Before you get started, ensure that you meet the following requirements:
+* **User Trust**: Builds confidence among users that their interactions with the app are secure, which is essential for user adoption and engagement.
 
-</br>
-<details><summary id="oad">1. OpenAPI Description (OAD) </summary>
+You can enable the following authentication methods for your API-based message extension:
 
-Ensure that you adhere to following guidelines for OpenAPI Description (OAD) document:
+**None**: If your message extension doesn’t require any authentication for the user to access the API, you can set the authType to “none” in your app manifest.
 
-* OpenAPI versions 2.0 and 3.0.x are supported.
-* JSON and YAML are the supported formats.
-* The request body, if present, must be application/Json.
-* Define an HTTPS protocol server URL for the `servers.url` property.
-* Only POST and GET HTTP methods are supported.
-* The OpenAPI Description document must have an `operationId`.
-* Only one required parameter without a default value is allowed.
-* A required parameter with a default value is considered optional.
-* Users must not enter a parameter for a header or cookie.
-* The operation must not have a required header or cookie parameters without default values.
-* Ensure that there are no remote references in the OpenAPI Description document.
-* Constructing arrays for the request isn’t supported; however, nested objects within a JSON request body are supported.
-* Teams doesn't support the `oneOf`, `anyOf`, `allOf`, and `not` (swagger.io) constructs.
+**Secret Service Auth**: For a more secure method, you can implement secret service authentication, which uses a secret token known only to the app and the API service to authenticate requests.
```

Comment:
make secret service auth and entra auth link to respective dedicated pages

Reply:
Added link as suggested.